tensorflow-utils/src/main/java/org/tensorflow/util/StaticApi.java

```diff
@@ -0,0 +1,708 @@
+   * @throws IllegalArgumentException if shape is null, has unknown dimensions or has size bigger
+   *                                  in the buffer size
+   */
+  static DoubleNdArray ndArrayOf(DoubleDataBuffer buffer, Shape shape) {
+    return NdArrays.wrap(buffer, shape);
+  }
+
+  /**
+   * Wraps a boolean buffer in an N-dimensional array of a given shape.
+   *
+   * @param buffer buffer to wrap
+   * @param shape shape of the array
+   * @return new boolean N-dimensional array
+   * @throws IllegalArgumentException if shape is null, has unknown dimensions or has size bigger
+   *                                  in the buffer size
+   */
+  static BooleanNdArray ndArrayOf(BooleanDataBuffer buffer, Shape shape) {
+    return NdArrays.wrap(buffer, shape);
+  }
+
+  /**
+   * Creates scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new scalar
+   */
+  static <T> NdArray<T> scalarOfObject(T value) {
+    return NdArrays.scalarOfObject(value);
+  }
+
+  /**
+   * Creates byte scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new byte scalar
+   */
+  static ByteNdArray scalarOf(byte value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates short scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new short scalar
+   */
+  static ShortNdArray scalarOf(short value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates int scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new int scalar
+   */
+  static IntNdArray scalarOf(int value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates long scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new long scalar
+   */
+  static LongNdArray scalarOf(long value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates float scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new float scalar
+   */
+  static FloatNdArray scalarOf(float value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates double scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new double scalar
+   */
+  static DoubleNdArray scalarOf(double value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates boolean scalar (rank 0) initialized with the given value.
+   *
+   * @param value scalar value
+   * @return new boolean scalar
+   */
+  static BooleanNdArray scalarOf(boolean value) {
+    return NdArrays.scalarOf(value);
+  }
+
+  /**
+   * Creates a vector (rank 1) initialized with the given values.
+   *
+   * @param values vector values
+   * @return new vector
+   * @throws IllegalArgumentException if values is null
+   */
+  static <T> NdArray<T> vectorOfObjects(T... values) {
```

Comment:
This is just a wrapper around references right? Is it clear that we need that kind of loose bound? A multidimensional array of references doesn't give us the locality or performance that we're looking for, but people might think it will.

Reply:
This vector won't be performant, true, but a tensor of strings is a legitimate multidimensional array though and this endpoint is mainly there to support these tensors. 

For example, since you can do:
```
TInt32.ofShape(2, 2)
    .set(vectorOf(10, 20), 0)
    .set(vectorOf(30, 40), 1);
```
ideally you should also be able to do:
```
TString.ofShape(2, 2)
    .set(vectorOf("a", "b"), 0)
    .set(vectorOf("c", "d"), 1);
```
But there is another problem: Because of how a tensor of string is structured internally, its data must be known at allocation time and has to be read-only after (as opposed to other type of tensors). Thus, we need to build up the N-dimensional array in memory first and then pass it at tensor creation, like this:
```
TString.copyOf(ndArrayOf(String.class, shapeOf(2, 2))
    .set(vectorOf("a", "b"), 0)
    .set(vectorOf("c", "d"), 1)
);
```
So in the end, as you can see, we need those non-performant reference multidimensional arrays. 

I was  forced to add the `Objects` suffix to the name of some methods to avoid clashes with those of a primitive type. Are you suggesting that maybe we make these names even more explicit about the nature of the created data structure? Like `ndArrayOfReferences`, `vectorOfReferences`, etc.?

Reply:
Ok. I'd go with `somethingOfReferences` to make it clear, no preference between ndArray or vector.

Reply:
Ok, I did not liked having `*References` or `*Refs` suffix in the name of our methods, sounds a bit awkward, so I replaced this by `*Objects` and the upcoming documentation will be self-explanatory to explain the layout of those buffers and their impact on performances.